src/plugin/debugger/mhpdbgc.c

```diff
@@ -981,6 +983,22 @@ static void mhp_mcbs(int argc, char *argv[])
   } else {
     mhp_printf("MCB chain corrupt - missing final entry\n");
   }
+
+  // UMBs
+  for (first = 1, seg = 0x9fff, mcb = MK_FP32(seg, 0); mcb->id == 'M'; /* */) {
```

Comment:
Can we avoid hardcoding 0x9fff?
It won't work with non-default `$_dosmem`.
FreeDOS puts this address into LoL:66h,
which seems to be a non-standard location.
How about just looking past the first 'Z' mcb?
It should point exactly to this one. We can
check if there is an `M` mcb after the first 'Z'
mcb, which will mean UMB.

Reply:
I updated the commit, now finds it at 0x9fff with default values. With dosmem = 600, FDPP fails to find command.com and output of `mcbs` command is
~~~
dosdebug> mcbs
dosdebug> 
ADDR      PARAS  OWNER
02a3:0000 0x0016 [DOS]
02ba:0000 END

ADDR(UMA) PARAS  OWNER
95ff:0000 0x3700 [DOS]
MCB chain corrupt - missing final entry in UMA
~~~

Is that to be expected?